aioesphomeapi/reconnect_logic.py

```diff
@@ -226,11 +226,11 @@ async def _handle_connection_failure(self, err: Exception) -> None:
 
     def _schedule_connect(self, delay: float) -> None:
         """Schedule a connect attempt."""
-        self._cancel_connect("Scheduling new connect attempt")
```

Comment:
This was unconditionally cancelling the connection attempt in the middle of a handshake